gateway-ha/src/main/java/io/trino/gateway/ha/router/RoutingManager.java

```diff
@@ -133,13 +134,13 @@ public void updateBackEndHealth(String backendId, Boolean value)
     public void updateBackEndStats(List<ClusterStats> stats)
     {
         for (ClusterStats clusterStats : stats) {
-            updateBackEndHealth(clusterStats.clusterId(), clusterStats.healthy());
+            updateBackEndHealth(clusterStats.clusterId(), clusterStats.healthState() == TrinoHealthStateType.HEALTHY);
         }
     }
 
     /**
      * This tries to find out which backend may have info about given query id. If not found returns
-     * the first healthy backend.
+     * the first healthState backend.
```

Comment:
`the first healthState backend` looks wrong grammar.

Reply:
+1 on the chain

- the usage of 'healthy' here is okay as it's relating to the backend manager in most cases which is still a boolean healthy/unhealthy
- 'healthy' is now one of our states so you can say things like `first healthy backend` / `first pending backend` / `first unhealthy backend` etc (when leveraging the healthState variable that is, which is unrelated to this current usage)

Reply:
agreed. It's a result of refactoring. I'll fix it